src/Persistent_cohomology/include/gudhi/Persistent_cohomology.h

```diff
@@ -172,11 +166,17 @@ class Persistent_cohomology {
    * valid. Undefined behavior otherwise. */
   void compute_persistent_cohomology(Filtration_value min_interval_length = 0) {
     interval_length_policy.set_length(min_interval_length);
+    Simplex_key idx_fil = -1;
+    std::vector<Simplex_key> vertices; // so we can check the connected components at the end
     // Compute all finite intervals
     for (auto sh : cpx_->filtration_simplex_range()) {
+      cpx_->assign_key(sh, ++idx_fil);
+      dsets_.make_set(cpx_->key(sh));
       int dim_simplex = cpx_->dimension(sh);
+      if (cpx_->filtration(sh) == std::numeric_limits<Filtration_value>::infinity()) break;
```

Comment:
As @mglisse was suggesting:
```suggestion
      if (std::numeric_limits<Filtration_value>::has_infinity &&
          cpx_->filtration(sh) == std::numeric_limits<Filtration_value>::infinity()) break;
```